Trouble with no-unused-vars rule

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

v8.32.0

### What version of `eslint-plugin-astro` are you using?

0.21.1

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "extends": [
    "plugin:astro/recommended"
  ],
  "rules": {
    "no-unused-vars": "error"
  },
  "overrides": [
    {
      "files": [
        "*.astro"
      ],
      "parser": "astro-eslint-parser",
      "parserOptions": {
        "extraFileExtensions": [
          ".astro"
        ],
        "ecmaVersion": "latest"
      }
    }
  ]
}
```
</details>

```astro
---
import SomeComponent from '../components/SomeComponent.astro';
---

<SomeComponent />
```


### What did you expect to happen?

I was hoping that the Astro plugin would work with the [`no-unused-vars` rule](https://eslint.org/docs/latest/rules/no-unused-vars).

### What actually happened?

```
/path/to/src/pages/problem.astro
  2:8  error  'SomeComponent' is defined but never used  no-unused-vars
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-nzoac8?file=README.md

### Additional comments

Much like the [`eslint-plugin-react`](https://www.npmjs.com/package/eslint-plugin-react) allows ESLint rules (such as `no-unused-vars`) to work with JSX, I was hoping that this plugin would allow the same for Astro components.  I see that there are [integration tests](https://github.com/ota-meshi/eslint-plugin-astro/blob/3e26b95340ce7051d8efd1dbdd1e7b67cbdffcc8/tests/src/integration/no-unused-vars.ts) using the `no-unused-vars` rule, but it looks like those don't cover the use of a variable from a component script in a component template.

--- a/.changeset/real-games-hunt.md
+++ b/.changeset/real-games-hunt.md
@@ -1,0 +1,5 @@
+---
+"eslint-plugin-astro": minor
+---
+
+update astro-eslint-parser

--- a/package.json
+++ b/package.json
@@ -54,7 +54,7 @@
   "dependencies": {
     "@jridgewell/sourcemap-codec": "^1.4.14",
     "@typescript-eslint/types": "^5.25.0",
-    "astro-eslint-parser": "^0.9.0",
+    "astro-eslint-parser": "^0.10.1",
     "eslint-utils": "^3.0.0",
     "postcss": "^8.4.14",
     "postcss-selector-parser": "^6.0.10"

